### PR TITLE
SEARCH-1968: Static tests to check for dependency of Elasticsearch modules

### PR DESCRIPTION
**Pull request by RuslanKostiv1**
### Description (*)
Add tests to check that no direct dependencies are added on Elasticsearch modules by any other module.

### Related Pull Requests
https://github.com/magento-commerce/magento2ce/pull/7570
https://github.com/magento-commerce/magento2ee/pull/3270

### Fixed Issues (if relevant)

1. Fixes SEARCH-1968: Static tests to check for dependency of Elasticsearch modules

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by RuslanKostiv1** [comment]
@andrewbess @xmav @eduard13 @sidolov @coderimus  please review the PR, see related PRs for more details

File: dev/tests/static/testsuite/Magento/Test/Integrity/_files/extension_dependencies_test/extension_conflicts/inventory.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+return [
+    // the following modules must be disabled when Live Search is used
+    // so core modules must not be dependent on them
+    'Magento\LiveSearch' => [
+        'Magento\InventoryElasticSearch',
```
Comment (andrewbess) [inline dev/tests/static/testsuite/Magento/Test/Integrity/_files/extension_dependencies_test/extension_conflicts/inventory.php]: Please fix module naming

```suggestion
        'Magento\InventoryElasticsearch',
```

**Review by andrewbess — CHANGES_REQUESTED** [review]
Hello @RuslanKostiv1 
Thank you for your contribution
Please check my suggestion
Also, I think it would be better to import this PR in `magento-commerce` org

**Comment by RuslanKostiv1** [comment]
Created PR for commerce instead

**Pull request closed without merging**